model/project_ref.go

```diff
@@ -1255,6 +1304,180 @@ func (projectRef *ProjectRef) Upsert() error {
+					ProjectRefFilesIgnoredFromCacheKey:   1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefAccessSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					ProjectRefPrivateKey:    1,
+					ProjectRefRestrictedKey: 1,
+					ProjectRefAdminsKey:     1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefVariablesSection:
+		err = db.Update(ProjectVarsCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectVarsMapKey:    1,
+					privateVarsMapKey:    1,
+					restrictedVarsMapKey: 1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefGithubSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectRefPRTestingEnabledKey:      1,
+					projectRefGithubChecksEnabledKey:   1,
+					projectRefGithubTriggerAliasesKey:  1,
+					projectRefGitTagVersionsEnabledKey: 1,
+					ProjectRefGitTagAuthorizedUsersKey: 1,
+					ProjectRefGitTagAuthorizedTeamsKey: 1,
+					projectRefCommitQueueKey:           1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+		catcher := grip.NewBasicCatcher()
+		for _, a := range before.Aliases {
+			// remove only internal aliases
+			if utility.StringSliceContains(evergreen.InternalAliases, a.Alias) {
+				catcher.Add(RemoveProjectAlias(a.ID.Hex()))
+			}
+		}
+		if catcher.HasErrors() {
+			// log the changes before returning whatever errors were caught
+			_ = getAndLogProjectModified(p.Id, caller, before)
+			return errors.Wrapf(catcher.Resolve(), "error defaulting to repo for section '%s'", section)
+		}
+
+	case ProjectRefNotificationsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{projectRefNotifyOnFailureKey: 1},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+		// handle subscriptions
+		catcher := grip.NewBasicCatcher()
+		for _, sub := range before.Subscriptions {
+			catcher.Add(event.RemoveSubscription(sub.ID))
+		}
+		if catcher.HasErrors() {
+			// log the changes before returning whatever errors were caught
+			_ = getAndLogProjectModified(p.Id, caller, before)
+			return errors.Wrapf(catcher.Resolve(), "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefAliasSection:
+		catcher := grip.NewBasicCatcher()
+		// remove only non-internal aliases
+		for _, a := range before.Aliases {
+			if !utility.StringSliceContains(evergreen.InternalAliases, a.Alias) {
+				catcher.Add(RemoveProjectAlias(a.ID.Hex()))
```

Comment:
This seems to be doing the same thing that's done on 1384. Should this be specific to the Patch Aliases section (as opposed to project aliases)?

Reply:
This is doing the _opposite_ thing as 1384: removes the project alias if it's _not_ an internal alias. Patch Aliases are still project aliases; they are defined at the project level and can be used for patch creation. For example our test alias: https://evergreen.mongodb.com/projects##evergreen

Reply:
haha yes, I missed that! thanks!

Reply:
Would it be possible to update the comment above? I think the comment is the same.

Reply:
nope, that's not true. never mind :)

Reply:
Would it maybe be possible to elaborate in the comment that non-internal aliases represent patch aliases?